lib/common/object.ts

```diff
@@ -0,0 +1,29 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+export function isPlainObject(item: unknown): boolean {
+  return item !== null && !Array.isArray(item) && Object.prototype.toString.call(item) === '[object Object]';
+}
+
+export function mergeDeep(target: any, ...sources: any): object {
+  if (!sources.length) return target;
+
+  for (const source of sources) {
+    if (isPlainObject(target) && isPlainObject(source)) {
+      for (const key in source) {
+        if (isPlainObject(source[key])) {
+          if (!target[key]) target[key] = {};
```

Comment:
I'm surprised prettier or the linter doesn't complain, I thought we didn't authorize one-liner like these.